src/pretix/base/payment.py

```diff
@@ -175,6 +175,15 @@ def abort_pending_allowed(self) -> bool:
         """
         return False
 
+    def create_invoice_immediately(self, order):
+        """
+        Whether or not to create an invoice right after this payment method has been chosen for ``order``.
+        Invoices send on succeeded payment are dealt with separately.
+        By default, this is True only if the event is configured to generate invoices for all orders immediately,
+        but it might be overwritten for e.g. bank transfer.
+        """
+        return self.event.settings.get('invoice_generate') == 'True'
```

Comment:
This is more of a general design question: I'm not sure if I like moving this logic into the payment provider. It creates issues where e.g. a invoice is no longer generated if an order currrently has no open payment (see below) and I imagine it will be hard to maintain and explain to users going forward if plugins to crazy stuff here.

I'd rather have this function to be just ``return False`` by default, and keep the logic in the pretix core, put like this (pseudocode):

```
# in services/orders.py:
if self.event.settings.get('invoice_generate') == 'True' or (self.event.settings.get('invoice_generate') == 'paid' and payment_provider.create_invoice_immediately(order)):
    # generate invoice
```

I'd then also change the settings label of ``invoice_generate = paid`` to something like "Automatically after payment, or during payment if required by payment method"